### PR TITLE
Only upload an extended bolus when it is complete

### PR DESCRIPTION
**Pull request by pazaan**
* Fixes https://trello.com/c/scIIYbAi

### CODE REVIEW

**Comment by pazaan** [comment]
[[600-series Bug] Can’t upload while an extended bolus is in progress](https://trello.com/c/scIIYbAi/420-600-series-bug-cant-upload-while-an-extended-bolus-is-in-progress)

File: lib/drivers/medtronic600/NGPHistoryParser.js [hunk 1]
```diff
@@ -1100,17 +1100,13 @@ class NGPHistoryParser {
   findMatchingDualBolusEvent(bolus) {
     const matchingBolusIndex = _.findIndex(this.events, event => event === bolus);
 
-    // It would be nice to use _.find here, but lodash 3.10.1 doesn't support fromIndex
-    for (let i = matchingBolusIndex; i < this.events.length; i++) {
-      const event = this.events[i];
-      if (event.eventType === NGPHistoryEvent.EVENT_TYPE.DUAL_BOLUS_PART_DELIVERED &&
+    return _.find(
```
Comment (gniezen) [inline lib/drivers/medtronic600/NGPHistoryParser.js]: Nice to see our dependency update work is helping you! :)

File: lib/drivers/medtronic600/NGPHistoryParser.js [hunk 2]
```diff
@@ -1743,36 +1739,40 @@ class NGPHistoryParser {
...
-            events.push(bolus.done());
+            this.addTimeFields(bolus, beginTimestamp);
+
+            if (event.isWizardBolus()) {
+              // Use parseFloat(num.toFixed(3)), because in JavaScript,
+              // 5.20 + 5.15 = 10.350000000000001
+              const wizardEvent = this.findWizardForBolus(
+                event,
+                parseFloat((event.normalProgrammedAmount + event.squareProgrammedAmount)
+                  .toFixed(3)),
```
Comment (gniezen) [inline lib/drivers/medtronic600/NGPHistoryParser.js]: I usually use 5 significant digits, as I think that's what we use elsewhere as well. Or is there a specific reason for using three digits here?

File: lib/drivers/medtronic600/medtronic600Simulator.js [hunk 3]
```diff
@@ -482,15 +488,15 @@ class Medtronic600Simulator {
             restoredBasal.deviceTime =
               sundial.formatDeviceTime(sundial.parseFromFormat(activeSuspend.deviceTime).getTime() +
               activeSuspend.duration);
-            if (restoredBasal.duration < 0) {
-              throw new Error('Should never get a duration of less than zero');
-            } else if (restoredBasal.duration > 0) {
+            Medtronic600Simulator.checkDuration(restoredBasal);
+            if (restoredBasal.duration > 0) {
```
Comment (gniezen) [inline lib/drivers/medtronic600/medtronic600Simulator.js]: What happens if duration is === 0? Is it just dropped? Should we log this to console or throw an error?

File: lib/drivers/medtronic600/medtronic600Simulator.js [hunk 4]
```diff
@@ -521,9 +528,8 @@ class Medtronic600Simulator {
             restoredSuspend.duration = Date.parse(restoredSuspend.basalEndTime) -
               Date.parse(restoredSuspend.time);
 
-            if (restoredSuspend.duration < 0) {
-              throw new Error('Should never get a duration of less than zero');
-            } else if (restoredSuspend.duration > 0) {
+            Medtronic600Simulator.checkDuration(restoredSuspend);
+            if (restoredSuspend.duration > 0) {
```
Comment (gniezen) [inline lib/drivers/medtronic600/medtronic600Simulator.js]: See previous comment regarding duration === 0

File: lib/drivers/medtronic600/NGPHistoryParser.js (hunk 2 above)
Comment (pazaan) [reply lib/drivers/medtronic600/NGPHistoryParser.js]: No particular reason. I didn't realise that 5 was used elsewhere.

**Comment by gniezen** [comment]
LGTM! :ship:

**Pull request merged**